refinery/core/tests.py

```diff
@@ -2297,3 +2305,151 @@ def test_create_workflow_engine_bad_group_name(self):
+
+    def test_users_created(self):
+        # + 2 instead of the expected 1 because the emission of
+        # the post_migrate signal creates the AnonymousUser
+        self.assertEqual(
+            self.site_statistics_a.users_created,
+            self.site_statistics_a._get_previous_instance().users_created + 2
+        )
+        self.assertEqual(self.site_statistics_b.users_created, 0)
+        self.assertEqual(
+            self.site_statistics_c.users_created,
+            self.site_statistics_c._get_previous_instance().users_created + 2
+        )
+
+    def test_groups_created(self):
+        self.assertEqual(self.site_statistics_a.groups_created, 0)
+        self.assertEqual(self.site_statistics_b.groups_created, 0)
+        self.assertEqual(self.site_statistics_c.groups_created, 1)
+
+    def test_unique_user_logins(self):
+        self.assertEqual(self.site_statistics_a.unique_user_logins, 1)
+        self.assertEqual(self.site_statistics_b.unique_user_logins, 0)
+        self.assertEqual(self.site_statistics_c.unique_user_logins, 2)
+
+    def test_total_user_logins(self):
+        self.assertEqual(self.site_statistics_a.total_user_logins, 1)
+        self.assertEqual(self.site_statistics_b.total_user_logins, 0)
+        self.assertEqual(self.site_statistics_c.total_user_logins, 4)
+
+    def test_total_workflow_launches(self):
+        self.assertEqual(self.site_statistics_a.total_workflow_launches, 0)
+        self.assertEqual(self.site_statistics_b.total_workflow_launches, 0)
+        self.assertEqual(self.site_statistics_c.total_workflow_launches, 1)
+
+    def test_total_visualization_launches(self):
+        self.assertEqual(
+            self.site_statistics_a.total_visualization_launches, 0)
+        self.assertEqual(
+            self.site_statistics_b.total_visualization_launches, 0)
+        self.assertEqual(
+            self.site_statistics_c.total_visualization_launches,
+            1
+        )
+
+    def test__get_previous_instance_returns_itself_if_only_instance(self):
+        SiteStatistics.objects.all().delete()
+
+        site_statistics = SiteStatistics.objects.create()
+        self.assertIsNotNone(site_statistics._get_previous_instance())
+        self.assertEqual(site_statistics._get_previous_instance(),
+                         site_statistics)
+
+    def test_periodic_task_is_scheduled_for_weekly_runs(self):
+        self.assertEqual(
+            settings.CELERYBEAT_SCHEDULE["collect_site_statistics"],
+            {
+                'task': 'core.tasks.collect_site_statistics',
+                'schedule': timedelta(days=1),
+                'options': {
+                    'expires': 30,  # seconds
+                }
+            }
+        )
+
+    def test_collect_site_statistics_creates_new_instance(self):
+        initial_site_statistics_count = SiteStatistics.objects.count()
+        with self.settings(CELERY_ALWAYS_EAGER=True):
```

Comment:
`CELERY_ALWAYS_EAGER` should not be necessary since `collect_site_statistics()` is called directly.